core/templates/store.area.template.html.tid

```diff
@@ -1,14 +1,30 @@
 title: $:/core/templates/store.area.template.html
 
-<$reveal type="nomatch" state="$:/isEncrypted" text="yes">
-`<div id="storeArea" style="display:none;">`
-<$list filter=<<saveTiddlerFilter>> template="$:/core/templates/html-div-tiddler"/>
-<$list filter={{{ [<skinnySaveTiddlerFilter>] }}} template="$:/core/templates/html-div-skinny-tiddler"/>
-`</div>`
-</$reveal>
-<$reveal type="match" state="$:/isEncrypted" text="yes">
-`<!--~~ Encrypted tiddlers ~~-->`
-`<pre id="encryptedStoreArea" type="text/plain" style="display:none;">`
-<$encrypt filter=<<saveTiddlerFilter>>/>
-`</pre>`
-</$reveal>
\ No newline at end of file
+\whitespace trim
+<!-- Unencrypted -->
+<$list filter="[[$:/isEncrypted]get[text]else[no]match[no]]">
+    <$list filter="[[storeAreaFormat]is[variable]getvariable[]else[json]match[json]]">
+        <!-- New-style JSON store area, with an old-style store area for compatibility with v5.1.x tooling -->
+        `<script class="tiddlywiki-tiddler-store" type="application/json">`
+            <$macrocall $name="jsontiddlers" filter=<<saveTiddlerFilter>> spaces="-1" escapeUnsafeScriptCharacters="yes" $output="text/raw"/>
```

Comment:
@Jermolene The `spaces="-1"` is very hacky. It causes problems, when I want to fix the "parseInt(spaces)" problem in the "jsontiddlers" macro. ... 

`JSON.stringify(value, replacer, space)` "space" can be a number OR a string eg: "\t"

`spaces` is used to transfer a completely unrelated information to `exports.getTiddlersAsJson = function(filter,options)` which already has an `option` that can be easily used to hold a new parameter. So imo there is no need to "misuse spaces" .. 

I would suggest, that the `jsontiddlers` macro gets a new parameter eg: `linebreakBetweenTiddlers`

Reply:
Hi @pmario at one point I did have a separate parameter for "linebreakBetweekTiddlers", but the problem is that it is not independent of the "spaces" setting; it doesn't make sense to have "spaces=3" and "linebreakBetweekTiddlers=true". In any event, can't "spaces=\t" coexist with "spaces=-1"?

Reply:
I mean apart from the parseInt() problem that you referred to

Reply:
Calling the macro looks like this atm

```
<$macrocall $name="jsontiddlers" filter=<<exportFilter>> spaces={{{[charcode[9]]}}} $output="text/raw"/>
```

I need to do this in "jsontiddlers" which is (kind of) ugly. ... spaces can be any string 

```
exports.run = function(filter,spaces,escapeUnsafeScriptCharacters) {
	var jsonString = (typeof spaces === "string") ? spaces : null,
	    jsonString = (spaces === "-1") ? null : jsonString,
	    json = this.wiki.getTiddlersAsJson(filter,{
	    spaces: (jsonString) ? jsonString : $tw.utils.parseInt(spaces),
```

Reply:
Hi @pmario why not just do this and let `this.wiki.getTiddlersAsJson` resolve the overloading of the spaces parameter?

```
exports.run = function(filter,spaces,escapeUnsafeScriptCharacters) {
	var json = this.wiki.getTiddlersAsJson(filter,{
		spaces: spaces,
		escapeUnsafeScriptCharacters: escapeUnsafeScriptCharacters === "yes"
	});
	return json;
};
```

Reply:
uups the code doesn't work. ... So I do need a new parameter to get it going. .. What if the `linebreakBetweekTiddlers` option overwrites the "spaces" parameter. .. We do have many options, that take precedence over other params.